app/src/main/java/org/glucosio/android/activity/ExternalLinkActivity.java

```diff
@@ -89,13 +89,20 @@ public void setupToolbarTitle(String link) {
     }
 
     @Override
-    public String extractAction() {
-        Bundle p;
+    public String extractTitle() {
+        String title = "";
         if (getIntent().getExtras() != null) {
-            p = getIntent().getExtras();
-            return p.getString("key");
+            title = getIntent().getStringExtra(TITLE_KEY);
         }
-        return null;
+        return title;
+    }
+
+    @Override public String extractUrl() {
```

Comment:
You have two methods `extractUrl` and `extractTitle` which can be replaced with one `getIntentExtraSafe(@NonNull String extraKey)`